src/submission/exceptions/exceptions.ts

```diff
@@ -260,3 +274,67 @@ export const checkForProgramAndEntityExceptions = async ({
 	});
 	return { filteredErrors, normalizedRecord };
 };
+
+/**
+ * Collect all exception records related to a set of Donors
+ *
+ * @param programId
+ * @param filters
+ * @returns [ ExceptionRecords ]
+ */
+export async function getExceptionManifestRecords(
+	programId: string,
+	filters: { donorIds: number[]; submitterDonorIds: string[] },
+): Promise<ExceptionManifestRecord[]> {
+	const { donorIds, submitterDonorIds: querySubmitterIds } = filters;
+
+	const donorsByDonorId = await getDonorsByIds(donorIds);
+	const donorsBySubmitterId = await Promise.all(
+		querySubmitterIds.map(
+			async (submitterId) => await findDonorBySubmitterId(submitterId, programId),
+		),
+	);
+
+	const donors = [...donorsByDonorId, ...donorsBySubmitterId].filter(notEmpty).filter(
+		(donorRecord, index, donorArray) =>
+			// Filter duplicates
+			index === donorArray.findIndex((donor) => donor.donorId === donorRecord.donorId),
+	);
+
+	// Exceptions only store submitterIds, so all submitterIds have to be collected before we can filter exceptions
+	const submitterDonorIds = donors.map((donor) => donor.submitterId);
+
+	const { programException, entityException } = await queryForExceptions(programId);
+
+	const missingEntityException = await getByProgramId(programId);
```

Comment:
I'm unclear here what the difference between an `entityException` and a `missingEntityException` is.
There may be naming to tweak overall and this work is bringing out that improvement.
For here even some comments would help.

for `queryForExceptions` and `getByProgramId` it may be beneficial to import an object and then call the method, so more context is given, something like: `	const missingEntityException = await missingExceptionAPI.getByProgramId(programId);`

Reply:
We can alias this as `entityPropertyException`